app/uk/gov/hmrc/tai/connectors/TaxAccountConnector.scala

```diff
@@ -43,20 +42,28 @@ class TaxAccountConnector @Inject()(
   featureTogglesConfig: FeatureTogglesConfig)(implicit ec: ExecutionContext)
     extends HodsSource {
 
+  lazy val uuid = UUID.randomUUID().toString
+
+  def hcWithDesHeaders(implicit hc: HeaderCarrier): HeaderCarrier = createHeader.withExtraHeaders(
+    "Gov-Uk-Originator-Id" -> desConfig.originatorId,
+    HeaderNames.xSessionId -> hc.sessionId.fold("-")(_.value),
+    HeaderNames.xRequestId -> hc.requestId.fold("-")(_.value)
+  )
+
   def hcWithHodHeaders(implicit hc: HeaderCarrier): HeaderCarrier =
     if (featureTogglesConfig.desEnabled) {
-      createHeader.withExtraHeaders("Gov-Uk-Originator-Id" -> desConfig.originatorId)
+      hcWithDesHeaders
     } else {
-      hc.withExtraHeaders("Gov-Uk-Originator-Id" -> npsConfig.originatorId)
+      hc.withExtraHeaders("Gov-Uk-Originator-Id" -> npsConfig.originatorId, "CorrelationId" -> uuid)
```

Comment:
Missing session and request IDs

Reply:
Session and request Ids are coming with incoming hc itself . Here we are adding extra headers for orgininator and correlation id